bc/channel/models.py

```diff
@@ -182,12 +181,13 @@ def validate_access_token(self):
             )
             raise e
         finally:
-            if not lock.owned():
-                return
-            try:
-                lock.release()
-            except Exception as e:
-                logger.error(f"Error releasing lock for channel {self}:\n{e}")
+            if lock.owned():
+                try:
+                    lock.release()
+                except Exception as e:
+                    logger.error(
+                        f"Error releasing lock for channel {self}:\n{e}"
+                    )
```

Comment:
```
bc/channel/models.py:186:17: B012 `return` inside `finally` blocks cause exceptions to be silenced
    |
184 |         finally:
185 |             if not lock.owned():
186 |                 return
    |                 ^^^^^^ B012
187 |             try:
188 |                 lock.release()
    |
```

https://docs.astral.sh/ruff/rules/jump-statement-in-finally/

This construct is actually deprecated from Python 3.14 due to its error-inducing potential: https://docs.python.org/3.14/whatsnew/3.14.html#pep-765-disallow-return-break-continue-that-exit-a-finally-block 

The fix here avoids the `return` by moving the `lock.release()` call under the `if lock.owned()` condition